src/backend/distributed/operations/stage_protocol.c

```diff
@@ -177,6 +178,17 @@ master_create_empty_shard(PG_FUNCTION_ARGS)
 		attemptableNodeCount = ShardReplicationFactor;
 	}
 
+	/* check if we are trying to create an empty placement on the coordinator */
```

Comment:
This error check seems a little implicit. It is not trivial for the readers of the code that `workerNodeListLength + 1 == ShardReplicationFactor` means we are not creating the placement on the coordinator. Also, do we strictly need this check? Wouldn't we end up the check in `CreateAppendDistributedShardPlacements` if we cannot all the necessary placements? 

```C

	/* check if we created enough shard replicas */
	if (placementsCreated < replicationFactor)
	{
		ereport(ERROR, (errmsg("could only create %u of %u of required shard replicas",
							   placementsCreated, replicationFactor)));
	}
```



Overall, to keep the changes small, would it make sense to have a check on `CreateAppendDistributedShardPlacements ()` such that `if (NodeIsCoordinator(workerNode))`, Citus skips (e.g., `continue`) creating placements on the node, and giving a `NOTICE` level message stating that creating placements for the append partitioned tables on the coordinator is not allowed. And, if there not enough placements that are created, Citus would throw and error as stated above.

Reply:
Makes sense. The idea was to explain the error further when there aren't enough shard replicas created because the coordinator is a worker node, so the `NOTICE` you suggested does the work with smaller changes (it is also more readable).